### PR TITLE
Powrng 21

### PR DESCRIPTION
**Pull request by michalbarczyk**

### CODE REVIEW

File: src/main/java/com/uxtest/backend/controller/QuestionnaireController.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    @GetMapping
+    public List<QuestionnaireDTO> getAllQuestionnaires() {
+
+        return questionnaireService.getAllQuestionnaires().stream()
+                .map(Questionnaire::toDTO)
+                .collect(Collectors.toList());
+    }
+
+//    @GetMapping("/byname")
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/controller/QuestionnaireController.java:39]: Unnecessary comment. Clean

File: src/main/java/com/uxtest/backend/dto/QuestionDTO.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+public class QuestionDTO {
+
+    @JsonProperty
+    private String content;
+
+    public String getContent() {
+        return content;
+    }
+
+    public Question toQuestion() {
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/dto/QuestionDTO.java:23]: Change name to parseQuestion

File: src/main/java/com/uxtest/backend/dto/QuestionDTO.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        return Question.builder()
+                .content(content)
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/dto/QuestionDTO.java:25]: use getter to content

File: src/main/java/com/uxtest/backend/dto/QuestionDTO.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class QuestionDTO {
+
+    @JsonProperty
+    private String content;
+
+    public String getContent() {
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/dto/QuestionDTO.java:19]: @Data gives you getter. Clean this

File: src/main/java/com/uxtest/backend/dto/QuestionnaireDTO.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+@NoArgsConstructor
+@AllArgsConstructor
+public class QuestionnaireDTO {
+
+    @JsonProperty
+    private String name;
+    @JsonProperty
+    private List<QuestionDTO> questions;
+
+    public String getName() {
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/dto/QuestionnaireDTO.java:28]: @Data gives you getter. Clean.

File: src/main/java/com/uxtest/backend/dto/QuestionnaireDTO.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        return name;
+    }
+
+    public List<QuestionDTO> getQuestions() {
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/dto/QuestionnaireDTO.java:32]: @Data gives you getter. Clean.

File: src/main/java/com/uxtest/backend/dto/QuestionnaireDTO.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        return questions;
+    }
+
+    public Questionnaire toQuestionnaire() {
+
+       return Questionnaire.builder()
+               .name(this.name)
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/dto/QuestionnaireDTO.java:39]: Use getters to name and questions

File: src/main/java/com/uxtest/backend/dto/QuestionnaireDTO.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+    public String getName() {
+        return name;
+    }
+
+    public List<QuestionDTO> getQuestions() {
+        return questions;
+    }
+
+    public Questionnaire toQuestionnaire() {
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/dto/QuestionnaireDTO.java:36]: change name to parseQuestionnaire

File: src/main/java/com/uxtest/backend/dto/UserDTO.java [hunk 9]
```diff
@@ -49,4 +49,4 @@ public User parseUser() {
     }
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/dto/UserDTO.java:49]: Garbage file

File: src/main/java/com/uxtest/backend/model/Question.java [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+    public String getContent() {
+        return content;
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public QuestionDTO toDTO() {
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/model/Question.java:45]: change name to mapToDTO

File: src/main/java/com/uxtest/backend/model/Question.java [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+    @OneToMany(mappedBy="question")
+    private List<Answer> answers;
+
+
+    public Question(String content) {
+        this.content = content;
+    }
+
+    public String getContent() {
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/model/Question.java:37]: @DaTa gives you getter. Clean.

File: src/main/java/com/uxtest/backend/model/Question.java [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+        return content;
+    }
+
+    public Long getId() {
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/model/Question.java:41]: @DaTa gives you getter. Clean.

File: src/main/java/com/uxtest/backend/model/Question.java [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+        return id;
+    }
+
+    public QuestionDTO toDTO() {
+        return QuestionDTO.builder()
+                .content(this.content)
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/model/Question.java:47]: use getter

File: src/main/java/com/uxtest/backend/model/Questionnaire.java [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+    private String name;
+
+    @OneToMany(mappedBy = "questionnaire")
+    private List<Question> questions;
+
+    public String getName() {
+        return name;
+    }
+
+    public Long getId() {
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/model/Questionnaire.java:39]: @DaTa gives you getter. Clean.

File: src/main/java/com/uxtest/backend/model/Questionnaire.java [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+        return id;
+    }
+
+    public List<Question> getQuestions() {
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/model/Questionnaire.java:43]: @DaTa gives you getter. Clean.

File: src/main/java/com/uxtest/backend/model/Questionnaire.java [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+        return questions;
+    }
+
+    public QuestionnaireDTO toDTO() {
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/model/Questionnaire.java:47]: Change name to mapToDTO

File: src/main/java/com/uxtest/backend/model/Questionnaire.java [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+        return QuestionnaireDTO.builder()
+                .name(this.name)
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/model/Questionnaire.java:49]: use getter to questions and name

File: src/main/java/com/uxtest/backend/service/QuestionnaireService.java [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+
+        questionnaireRepository.save(questionnaire);
+
+        for (var question : questionnaire.getQuestions()) {
+            question.setQuestionnaire(questionnaire);
+            questionRepository.save(question);
+        }
+    }
+
+//    public Optional<Questionnaire> getQuestionnaireByName(String name) {
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/service/QuestionnaireService.java:43]: Unnecessary comment

File: src/main/java/com/uxtest/backend/service/QuestionnaireService.java [hunk 19]
```diff
@@ -0,0 +1,47 @@
...
+    public QuestionnaireService(QuestionnaireRepository questionnaireRepository,
+                                QuestionRepository questionRepository) {
+        this.questionnaireRepository = questionnaireRepository;
+        this.questionRepository = questionRepository;
+    }
+
+    public List<Questionnaire> getAllQuestionnaires() {
+
+        return questionnaireRepository.findAll();
+
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/service/QuestionnaireService.java:29]: So many newlines

File: src/main/java/com/uxtest/backend/service/UserService.java [hunk 20]
```diff
@@ -1,38 +0,0 @@
-package com.uxtest.backend.service;
-
-import com.uxtest.backend.model.user.User;
```
Comment (RafisSomeone) [inline src/main/java/com/uxtest/backend/service/UserService.java:3]: merging this delete UserService please merge master to current

File: src/main/java/com/uxtest/backend/dto/UserDTO.java (hunk 9 above)
Comment (michalbarczyk) [reply src/main/java/com/uxtest/backend/dto/UserDTO.java:49]: Should I remove it? It means that model.user.User class has to be removed also

Comment (RafisSomeone) [reply src/main/java/com/uxtest/backend/dto/UserDTO.java:49]: My bad. Remove new lines added

**Pull request merged**